operating_systems.go

```diff
@@ -281,3 +300,60 @@ func (p *UserAgent) OS() string {
 func (p *UserAgent) Localization() string {
 	return p.localization
 }
+
+// Returns combined information for the operating system.
+func (p *UserAgent) OSInfo() OSInfo {
+	// Special case for iPhone weirdness
+	os := strings.Replace(p.os, "like Mac OS X", "", 1)
+	os = strings.Replace(os, "CPU", "", 1)
+	os = strings.Trim(os, " ")
+
+	osSplit := strings.Split(os, " ")
+
+	// Special case for x64 edition of Windows
+	if os == "Windows XP x64 Edition" {
+		osSplit = osSplit[:len(osSplit)-2]
+	}
+
+	var name, version string
+	if len(osSplit) == 1 {
+		name = osSplit[0]
+		version = ""
+	} else {
+		nameSplit := osSplit[:len(osSplit)-1]
+		version = osSplit[len(osSplit)-1]
+
+		// Nicer looking Mac OS X
+		if len(nameSplit) >= 2 && nameSplit[0] == "Intel" && nameSplit[1] == "Mac" {
+			nameSplit = nameSplit[1:]
+		}
+		name = strings.Join(nameSplit, " ")
+
+		if strings.Contains(version, "x86") || strings.Contains(version, "i686") {
+			// x86_64 and i868 are not Linux versions but architectures
+			version = ""
+		} else if version == "X" && name == "Mac OS" {
+			// X is not a version for Mac OS.
+			name = name + " " + version
+			version = ""
+		}
+	}
```

Comment:
I think it would be better to put this specific if-else block in another function.